graphs/graphs_floyd_warshall.py

```diff
@@ -1,62 +1,112 @@
 # floyd_warshall.py
 """
-The problem is to find the shortest distance between all pairs of vertices in a
-weighted directed graph that can have negative edge weights.
+The problem is to find and return the shortest distance between all pairs of vertices
+in a weighted directed graph that can have negative edge weights.
 """
 
 
-def _print_dist(dist, v):
-    print("\nThe shortest path matrix using Floyd Warshall algorithm\n")
-    for i in range(v):
-        for j in range(v):
-            if dist[i][j] != float("inf"):
-                print(int(dist[i][j]), end="\t")
-            else:
-                print("INF", end="\t")
-        print()
+def floyd_warshall(graph: list[list[float]], vertex: int) -> tuple:
+    # 1. For all edges from v to n, distance[i][j] = weight(edge(i, j)).
+
+    # 2. distance[i][j] = min(distance[i][j], distance[i][k] + distance[k][j]).
 
+    # 3. Step 2 is true for each pair of vertices.Repeat for k vertex in the graph.
+
+    # 4. Whenever distance[i][j] is given a new minimum value,
+    #   next vertex[i][j] = next vertex[i][k].
```

Comment:
This algorithm explanation should be moved into the docstring at the top of the file